src/shared/lib/timers/hooks/useDelayedInterval.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useRef } from 'react';
+
+import { useInterval, useOnForegroundDebounced, useOnceRef } from '../../';
+
+const Interval = 30000;
+
+const Delay = 5000;
+
+type Props = {
+  onTick: () => unknown;
+};
+
+const useDelayedInterval = ({ onTick }: Props) => {
+  const onTickRef = useRef(onTick);
+  onTickRef.current = onTick;
+
+  const onIntervalPass = useCallback(() => {
+    onTickRef.current();
+  }, []);
+
+  const { start, stop } = useInterval(onIntervalPass, Interval);
+
+  const delayTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  const startWithDelay = useCallback(() => {
+    if (delayTimeoutRef.current) {
+      clearTimeout(delayTimeoutRef.current);
```

Comment:
Please also perform `clearTimeout` on unmount. Otherwise, we may get a memory leak.

```
useEffect(() => () => {
  if (delayTimeoutRef.current) {
    clearTimeout(delayTimeoutRef.current);
  }
});
```

Reply:
ok